Packs/AtlassianConfluenceCloud/Integrations/AtlassianConfluenceCloud/AtlassianConfluenceCloud.py

```diff
@@ -1333,14 +1379,101 @@ def confluence_cloud_group_list_command(client: Client, args: Dict[str, str]) ->
         raw_response=response_json)
 
 
+def fetch_events(client: Client, last_run: dict[str, Any], fetch_limit: int):
+    demisto.debug(f'Starting fetch_events with last_run: {last_run} and fetch_limit: {fetch_limit}')
+    end_date = int((time.time() - 5) * 1000)
+    last_end_date = last_run.get('end_date', 0)
+    start_date = last_end_date + 1 if last_end_date else end_date - 60000
+    next_link = last_run.get('next_link', '')
+    is_cleanup = bool(next_link)
+    total_length = 0
+
+    demisto.debug(f'Initial start_date: {start_date}, end_date: {end_date}, next_link: {next_link}, is_cleanup: {is_cleanup}')
+    while True:
+        response = run_fetch_mechanism(client, next_link, start_date, end_date)
+        events = response['results']
+        next_link = response['_links'].get('next', None)
+        total_length += len(events)
+
+        demisto.debug(f'Fetched {len(events)} events, total_length: {total_length}, next_link: {next_link}')
+        if total_length >= fetch_limit:
+            diff = total_length - fetch_limit
+            correct_last_index = len(events) - diff
+            events = events[:correct_last_index]
+
+            demisto.debug(f'Fetch limit reached. Adjusting events to {len(events)} and calculating next_link.')
+            if next_link:
+                parsed_next_link = urllib.parse.urlparse(next_link)
+                query_params = urllib.parse.parse_qs(parsed_next_link.query)
+                query_params['start'] = [str(fetch_limit)]
+                next_link = urllib.parse.urlunparse(
+                    parsed_next_link._replace(
+                        query=urllib.parse.urlencode(query_params, doseq=True)
+                    )
+                )
+            else:
+                next_link = URL_SUFFIX['NEXT_LINK_TEMPLATE'].format(end_date, PAGE_SIZE, fetch_limit, start_date)
+            demisto.debug(f'Yielding events and next_link: {next_link}')
+            yield events, next_link, end_date
+            break
+
+        if not next_link:
+            if not is_cleanup:
+                demisto.debug('No more next_link and not in cleanup mode. Yielding events and ending fetch.')
+                yield events, next_link, end_date
+                break
+
+            # last call cleaned up the previous batch, start new batch
+            demisto.debug('Finished cleanup mode. Starting the new batch.')
+            response = run_fetch_mechanism(client, None, start_date, end_date)
```

Comment:
why are we calling run_fetch_mechanism here? it should only be done in the beggining of the loop

Reply:
This starts a new batch of fetching and the current `events` are the last events from the previous batch. If we were to use `continue` instead of a new fetch, we would end up deleting all the events that were at the end of the previous batch (the ones we used the clean-up flag to get).

Reply:
lets set a sync to discuss